contrib/ckptfile/README

```diff
@@ -1,2 +1,47 @@
-This plugin serves as a template.  Please modify the given
-functions according to your needs.
+# 'ckptfile' plugin
+
+This plugin allows you to:
+
+ a) specify a file that DMTCP should checkpoint; and
+ b) specify the path on restart of a file if it wasn't checkpointed.
+
+The plugin reads a user-specified database at checkpoint time. The
+database is a simple text file that specifies the files that should
+be checkpointed, and the restart paths of files that shouldn't be
+checkpointed.
+
+The database file's format is defined below.
+
+   file:ckpt={1|0};rstrt=file
+   file-pattern:ckpt={1|0};rstrt=file-pattern
+
+Note that for a file that's checkpointed (ckpt=1), the restart path
+(rstrt=XXX) is ignored.
+
+## Background
+
+There are two scenarios:
+
+ 1) DMTCP checkpoints an open file:
+
+    This happens when a user asks DMTCP to checkpoint a file being used
+    by an application. In this case, DMTCP saves a copy of the file as
+    it existed at checkpoint time in a separate directory.
+
+    On restart, the saved copy of the file is restored at its original
+    location, the file pointers are restored, and the user application
+    can continue to read from there.
+
+ 2) DMTCP does not checkpoint an open file:
+
+    This happens when a user does not ask DMTCP to checkpoint a file
+    being used by an application. In this case, DMTCP does not save a
+    copy of the file at checkpoint time.
+
+    On restart, the file that the application was using at checkpoint
+    time is expected to be at its original location, and only the file
+    pointers are restored.
+
+## Usage
+
+    $ DMTCP_CKPTFILES_DATA=/path/to/ckptfiles.dat dmtcp_launch --with-plugin /path/to/libdmtcp_ckptfile.so ./a.out
```

Comment:
Let's put the `## Usage` comment and the following two lines at the beginning of this README file.